app/utils/get_filtered_records.py

```diff
@@ -116,39 +116,59 @@ def get_all_records(research_fields=False, include_disputed_regions=False,
 
         query = query.all()
         # Convert from sqlalchemy object to dict
-        query_dict = [ q._asdict() for q in query]
+        query_dict = [q._asdict() for q in query]
 
         return query_dict
 
-'''
-Filter are in the following format: 
-{ 
-  'age' : ['Youth (13-17)', 'All'],
-  'country' : ['United States']
-}
-
-Output: set of records represented by dicts
-'''
-
 
 def get_filtered_records(research_fields=False, filters=None, columns=None, include_disputed_regions=False,
                          sampling_start_date=None, sampling_end_date=None, include_subgeography_estimates=False,
-                         publication_start_date=None, publication_end_date=None, prioritize_estimates=True,
+                         publication_start_date=None, publication_end_date=None, estimates_subgroup='all',
                          prioritize_estimates_mode='dashboard', include_in_srma=False, unity_aligned_only=False,
                          include_records_without_latlngs=False):
+    # Get all records from the database as a list of records represented by dicts
     query_dicts = get_all_records(research_fields, include_disputed_regions, unity_aligned_only,
                                   include_records_without_latlngs)
+
     if query_dicts is None or len(query_dicts) == 0:
         return []
 
+    # If estimates_subgroup is 'estimate_prioritization', perform estimate prioritization
+    if estimates_subgroup == 'prioritize_estimates':
+        result_df = pd.DataFrame(query_dicts)
+        if include_subgeography_estimates:
+            prioritized_records = get_prioritized_estimates_without_pooling(result_df,
+                                                                            subgroup_var="Geographical area",
+                                                                            mode=prioritize_estimates_mode)
+        else:
+            prioritized_records = get_prioritized_estimates(result_df, mode=prioritize_estimates_mode)
+        # If records exist, clean dataframe
+        if not prioritized_records.empty:
+            # Convert from True/None to True/False
+            for col in prioritized_records.columns:
+                # Note purpose of this line is to check if the col in question is a boolean col
+                # Can't simply check the dtype because cols with True/None instead of
+                # True/False have dtype="object" instead of "bool"
+                if True in prioritized_records[col].values:
+                    prioritized_records[col] = prioritized_records[col].fillna(False)
+            prioritized_records = prioritized_records.fillna(np.nan).replace({np.nan: None})
+        query_dicts = prioritized_records.to_dict('records')
+
+    # If estimates_subgroup is 'primary_estimates', just return the primary estimate for each study
+    # Otherwise, estimates_subgroup = 'all' so just return all estimates
+    elif estimates_subgroup == 'primary_estimates':
```

Comment:
Hmm I was only certain that this block would work well if moved before the filtering step and not estimate prioritization. Can you double check the results of this change for me, thanks.

Reply:
well i didn't test the timing to see if it's more efficient to put here but in terms of results it still gives the accurate results when you move this block up

Reply:
i.e. when "all_estimates" is selected it returns ~13k records, otherwise for the other 2 options it's about 3k records

Reply:
all you are doing is doing all this logic right after get all records, converting query dicts to df, running this logic, then converting df back to query dicts and carrying on with a subset of the results